### PR TITLE
Add the selected App Engine library dependencies to the maven project

### PR DESCRIPTION
**Pull request by nbashirbello**
Part of fix for #803

### CODE REVIEW

**Comment by nbashirbello** [comment]
PR not complete.

**Comment by nbashirbello** [comment]
@elharo 
Selecting the Endpoints library adds nothing to the maven project. But selecting the App Engine API works (the correct dependencies are added to the project).

**Comment by nbashirbello** [comment]
I figured out what the problem is. The latest archetypes with the endpoints and appengine api properties are not in maven central yet. The lates hello-world archetype is from Oct 13.

PR is ready for review

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java [hunk 1]
```diff
@@ -67,9 +71,9 @@ protected void execute(IProgressMonitor monitor)
     // The project ID is currently necessary due to tool bugs.
     properties.put("application-id", artifactId);
     properties.put("useObjectify", "false");
-    properties.put("useEndpoints1", "false");
-    properties.put("useEndpoints2", "false");
-    properties.put("useAppEngineApi", "false");
+    properties.put("useEndpoints1", appEngineLibraryIds.contains("appengine-endpoints"));
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: This value needs to be a string, not a boolean. Don't ask me to explain why but I spent a day debugging that. :-( Ditto below.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java [hunk 2]
```diff
@@ -130,4 +134,11 @@ void setArchetype(Archetype archetype) {
     this.archetype = archetype;
   }
 
+  void setAppEngineLibraryIds(List<Library> libraries) {
+    appEngineLibraryIds = new LinkedList<>();
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: When in doubt, ArrayList, though a HashSet might work best here.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java [hunk 3]
```diff
@@ -67,9 +71,9 @@ protected void execute(IProgressMonitor monitor)
     // The project ID is currently necessary due to tool bugs.
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: Independent of this PR: is this comment still valid?

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java (hunk 2 above)
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: +1 for a set

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java (hunk 3 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: I think the comment is irrelevant now.

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java [hunk 4]
```diff
@@ -67,9 +70,9 @@ protected void execute(IProgressMonitor monitor)
...
     properties.put("application-id", artifactId);
     properties.put("useObjectify", "false");
-    properties.put("useEndpoints1", "false");
-    properties.put("useEndpoints2", "false");
-    properties.put("useAppEngineApi", "false");
+    properties.put("useEndpoints1", Boolean.toString(appEngineLibraryIds.contains("appengine-endpoints")));
+    properties.put("useEndpoints2", false);
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: you still need a string on this one

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java [hunk 5]
```diff
@@ -64,12 +67,11 @@ protected void execute(IProgressMonitor monitor)
     properties.put("appengine-version", appengineArtifactVersion);
     properties.put("gcloud-version", gcloudArtifactVersion);
     properties.put("useJstl", "true");
-    // The project ID is currently necessary due to tool bugs.
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java]: Please put this back. It still seems to be true.

**Pull request merged**